alicloud/resource_alicloud_kvstore_instance.go

```diff
@@ -0,0 +1,358 @@
+				Default:  "Redis",
+				ValidateFunc: validateAllowedStringValue([]string{
+					"Memcache",
+					"Redis",
+				}),
+			},
+			"vswitch_id": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+			},
+			"engine_version": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Default:  "2.8",
+				ValidateFunc: validateAllowedStringValue([]string{
+					"2.8",
+					"4.0",
+				}),
+			},
+			"connection_domain": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"port": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"private_ip": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"backup_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"security_ips": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudKVStoreInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	conn := client.rkvconn
+
+	request, err := buildKVStoreCreateRequest(d, meta)
+	if err != nil {
+		return err
+	}
+
+	resp, err := conn.CreateInstance(request)
+
+	if err != nil {
+		return fmt.Errorf("Error creating Alicloud db instance: %#v", err)
+	}
+
+	d.SetId(resp.InstanceId)
+
+	// wait instance status change from Creating to Normal
+
+	if err := client.WaitForRKVInstance(d.Id(), Normal, DefaultLongTimeout); err != nil {
+		return fmt.Errorf("WaitForInstance %s got error: %#v", Running, err)
+	}
+
+	// Adding security ips
+	if secIps, ok := d.GetOk("security_ips"); ok && secIps != nil {
+		request := r_kvstore.CreateModifySecurityIpsRequest()
+		request.SecurityIps = LOCAL_HOST_IP
+		request.SecurityIpGroupName = "default"
+		request.InstanceId = resp.InstanceId
+		if len(d.Get("security_ips").(*schema.Set).List()) > 0 {
+			request.SecurityIps = strings.Join(expandStringList(d.Get("security_ips").(*schema.Set).List())[:], COMMA_SEPARATED)
+		}
+
+		err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+			if _, err := conn.ModifySecurityIps(request); err != nil {
+				return resource.NonRetryableError(fmt.Errorf("Create security whitelist ips got an error: %#v", err))
+			}
+			return nil
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return resourceAlicloudKVStoreInstanceRead(d, meta)
+}
+
+func resourceAlicloudKVStoreInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	conn := client.rkvconn
+	d.Partial(true)
+
+	if d.HasChange("instance_class") {
+		request := r_kvstore.CreateModifyInstanceSpecRequest()
+		request.InstanceId = d.Id()
+		request.InstanceClass = d.Get("instance_class").(string)
+		if _, err := conn.ModifyInstanceSpec(request); err != nil {
```

Comment:
I think ModifyInstanceSpecRequest should add a parameter "EffectiveTime" and set it to "Immediately"

Reply:
It seems like without any update.